### PR TITLE
[Console][FrameworkBundle] Revised console header formatting

### PR DESCRIPTION
**Pull request by ro0NL**
| Q | A |
| --- | --- |
| Branch? | "master" |
| Bug fix? | no |
| New feature? | yes (ui change) |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | not yet |
| Fixed tickets | #19095 |
| License | MIT |
| Doc PR | reference to the documentation PR, if any |

Before/After
image

### CODE REVIEW

**Comment by iltar** [comment]
`app` is not really the kernel, just the name. In this case I'd expect to see `AppKernel`, would that be nicer to show?

**Comment by ro0NL** [comment]
If it's namespaced it can get messy.. but you're right; it's different information. Both are useful :')

What about;
`Symfony 3.2.0-DEV on \My\AppKernel(name: app, env: dev, debug: true)`

**Comment by iltar** [comment]
I personally prefer the verbosity flag but I don't know how intuitive this is.

**Comment by javiereguiluz** [comment]
Another solution would be to remove the `name: app` thing and to not add the kernel info either.

**Comment by iltar** [comment]
There might be a lot more info about the kernel you'd want to see...  What about simply adding the version + release date (if available) and making an application info command? Not sure if this would be interesting for the core, but it might show nice things such as which bundles are enabled.

**Comment by ro0NL** [comment]
@iltar i was just thinking about that.. with `--verbose` we can go really crazy;

image

**Comment by iltar** [comment]
I'm afraid that eventually it will come down to this: We need a command line WDT variant

**Comment by ro0NL** [comment]
Agree, this is still not great as a header line when used with `bin/console`. So what do we need to show for `-V` then? (assuming everything is available with `bin/console info` in some time).

Perhaps kernel info is a  bit too specific indeed =/

**Comment by iltar** [comment]
For now I'd say a limit of what's available in the `Kernel` constants. I would love to have more debugging info on the command line though, such as a CLI WDT. Maybe a nice feature for 3.2 (though I have no time to work on this).

**Comment by javiereguiluz** [comment]
Please, let's focus first on finishing this pull request about the console header. Then we can discuss about the `kernel:info` command in a separate issue. Thanks!

**Comment by ro0NL** [comment]
Done.

File: src/Symfony/Bundle/FrameworkBundle/Console/Application.php [hunk 1]
```diff
@@ -100,6 +100,18 @@ public function all($namespace = null)
...
+    /**
+     * {@inheritdoc}
+     */
+    public function getLongVersion()
+    {
+        return parent::getLongVersion()
+            .sprintf(
+                ' (env: <comment>%s</comment>, debug: <comment>%s</comment>)',
+                $this->kernel->getEnvironment(), $this->kernel->isDebug() ? 'true' : 'false'
+            );
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: This should be on one line.

Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: Can i go with `<comment>%s</>` btw (vs. `<comment>%s</comment>`) btw? Would improve readability :)

Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: IIRC we already do that in other places.

File: src/Symfony/Bundle/FrameworkBundle/Console/Application.php [hunk 2]
```diff
@@ -100,6 +100,15 @@ public function all($namespace = null)
         return parent::all($namespace);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getLongVersion()
+    {
+        return parent::getLongVersion()
+            .sprintf(' (env: <comment>%s</>, debug: <comment>%s</>)', $this->kernel->getEnvironment(), $this->kernel->isDebug() ? 'true' : 'false');
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: Can you move this on one line?

Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: I wasnt sure :sweat_smile:

**Comment by fabpot** [comment]
I would not remove the app name here. You can have several one and being able to have it with `-V` is important.

**Comment by fabpot** [comment]
Any news on this one?

**Comment by fabpot** [comment]
@ro0NL Still interested in finishing this one?

**Comment by ro0NL** [comment]
Yes. Try to work on it today. As soon as my laptop is connected :(

Op 19 jul. 2016 17:35 schreef "Fabien Potencier" <notifications@github.com>:

**Comment by ro0NL** [comment]
Back up :) @fabpot it's going to be `Symfony 3.2.0-DEV (kernel: app, env: dev, debug: true)` then.. where `kernel` refers the kernel name, not the kernel type.

I think it makes sense.. if #19278 is accepted we have more detailed information anyway.

**Comment by fabpot** [comment]
Tests should be updated as they do not pass with these changes.

**Comment by ro0NL** [comment]
Green :)

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Pull request closed without merging**